spec/paypal_express/remote/baid_spec.rb

```diff
@@ -256,32 +227,58 @@
     payment_infos[2].status.should == :PROCESSED
     payment_infos[2].gateway_error.should == 'Success'
     payment_infos[2].gateway_error_code.should be_nil
+    find_value_from_properties(payment_infos[2].properties, 'payment_processor_account_id').should == @payment_processor_account_id
   end
 
   it 'should generate forms correctly' do
     context = @plugin.kb_apis.create_context(@call_context.tenant_id)
-    fields  = @plugin.hash_to_properties(
-                 :order_id => '1234',
-                 :amount   => 12,
-              )
+    fields = @plugin.hash_to_properties(
+        :order_id => '1234',
+        :amount => 12,
+    )
 
     properties = @plugin.hash_to_properties(
-      :create_pending_payment => false
+        :create_pending_payment => false
     )
 
     form = @plugin.build_form_descriptor(@pm.kb_account_id, fields, properties, context)
 
     form.kb_account_id.should == @pm.kb_account_id
-    form.form_method.should   == 'GET'
+    form.form_method.should == 'GET'
     form.form_url.should start_with('https://www.sandbox.paypal.com/cgi-bin/webscr')
   end
 
-  private
+end
+
+describe Killbill::PaypalExpress::PaymentPlugin do
+
+  include ::Killbill::Plugin::ActiveMerchant::RSpec
+  include BuildPluginHelpers
+  include BaidHelpers
+
+  context 'baid test with a single account' do
+    # Share the BAID
+    before(:all) do
+      # delete once here because we need to keep the initial response for later tests to find the payment processor account id
+      ::Killbill::PaypalExpress::PaypalExpressResponse.delete_all
+      @payment_processor_account_id = 'default'
+      @plugin = build_start_paypal_plugin
+      baid_setup
+    end
+
+    include_examples 'baid_spec_common'
+  end
+
+  context 'baid tests with multiple accounts' do
+    # Share the BAID
+    before(:all) do
+      # delete once here because we need to keep the initial response for later tests to find the payment processor account id
+      ::Killbill::PaypalExpress::PaypalExpressResponse.delete_all
+      @payment_processor_account_id = 'paypal_test_account'
```

Comment:
No strong opinion but since you're starting to leverage all sorts of RSpec features... 😺 I believe `let` is preferred to instance variables.

For reference: http://stackoverflow.com/questions/5359558/when-to-use-rspec-let.